src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -12,29 +12,29 @@ internal static partial class ThrowHelper
         // If the exception source is this value, the serializer will re-throw as JsonException.
         public const string ExceptionSourceValueToRethrowAsJsonException = "System.Text.Json.Rethrowable";
 
-        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_MaxDepthMustBePositive(string parameterName)
+        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_MaxDepthMustBePositive(string? parameterName)
         {
             return GetArgumentOutOfRangeException(parameterName, SR.MaxDepthMustBePositive);
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static ArgumentOutOfRangeException GetArgumentOutOfRangeException(string parameterName, string message)
+        private static ArgumentOutOfRangeException GetArgumentOutOfRangeException(string? parameterName, string? message)
         {
             return new ArgumentOutOfRangeException(parameterName, message);
         }
 
-        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_CommentEnumMustBeInRange(string parameterName)
+        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_CommentEnumMustBeInRange(string? parameterName)
         {
             return GetArgumentOutOfRangeException(parameterName, SR.CommentHandlingMustBeValid);
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static ArgumentException GetArgumentException(string message)
+        private static ArgumentException GetArgumentException(string? message)
         {
             return new ArgumentException(message);
         }
 
-        public static void ThrowArgumentException(string message)
+        public static void ThrowArgumentException(string? message)
```

Comment:
Can we preemptively mark all the "Throw..." public APIs with `[DoesNotReturn]`? That way, we are consistent with the annotations rather than only picking some methods.

`ThrowArgumentException` overloads
`ThrowArgumentException_PropertyNameTooLarge`
`ThrowArgumentException_ValueTooLarge`
`ThrowArgumentException_ValueNotSupported`
`ThrowInvalidOperationException_NeedLargerSpan`
`ThrowInvalidOperationOrArgumentException`
`ThrowInvalidOperationException` overloads
`ThrowInvalidOperationException_DepthNonZeroOrEmptyJson`
`ThrowJsonReaderException`
`ThrowArgumentException_InvalidCommentValue`
`ThrowArgumentException_InvalidUTF8`
`ThrowArgumentException_InvalidUTF16`
`ThrowInvalidOperationException_ReadInvalidUTF16`
`ThrowInvalidOperationException_ReadInvalidUTF16`